hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestListBlob.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.azurebfs;
+
+
+import java.util.List;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Assume;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.azurebfs.services.AbfsClient;
+import org.apache.hadoop.fs.azurebfs.services.PrefixMode;
+import org.apache.hadoop.fs.azurebfs.utils.TracingContext;
+
+public class ITestListBlob extends
+    AbstractAbfsIntegrationTest {
+
+  public ITestListBlob() throws Exception {
+    super();
+  }
+
+  @Test
+  public void testListBlob() throws Exception {
+    AzureBlobFileSystem fs = getFileSystem();
+    assumeNonHnsAccountBlobEndpoint(fs);
+    int i = 0;
+    while (i < 10) {
+      fs.create(new Path("/dir/" + i));
```

Comment:
There is Path() in the parent test class, that will give a unique path.

Is the path the only reason to add this to pom.xml for sequential run ?

Reply:
Reason for doing that is I am adding spied abfsClient to the fileSystem being used. There are mockito verifications (for times invoked) on the spied client. If tests would run in parallel, the verification would get wrong.

Reply:
This can be achieved using org.mockito.Mockito.verify and sequential requirement wont be there. We should avoid making test sequential unless there is a real need as it contributes to overall test run time.

Reply:
This test doesn't depend on the spiedClient. There are other tests in the class which does verify on the spiedClient.

Reply:
The tests have been simplified, removed the sequential constraint in pom.xml

Reply:
Have simplied test-class, have removed the sequential constriant.